doc/source/management/innodb_fake_changes.rst

```diff
@@ -4,17 +4,29 @@
+
+The way prefetching can be implemented without :ref:`innodb_fake_changes_page`, in most of the cases is by replaying the statements from the relay log but then manually converting all ``COMMITs`` to ``ROLLBACKs`` . This has the caveat of introducing the extra locking that is caused by the locks that are taken by the statements which are being replayed. The locks taken by statements executed by the process which is doing the prefetching can also cause lock contention with events that the SQL thread is trying to apply. Another issue with doing rollback is that, when a transaction changes data, old versions of the data are written to the undo log buffer. During the rollback phase |InnoDB| then has to read old versions of the data corresponding to what it was before the change from the undo log buffer and move it back to the |InnoDB| data page. In case of large transactions, or a transaction that updates a lot of data, the rollback can be costly and can generate significant amount of I/O.
+
+Keeping in view the need of prefetching and the current caveats the :variable:`innodb_fake_changes` variable was implemented. The :variable:`innodb_fake_changes` variable enables an option for the server-side which allows for prefetching to work in a more performant manner. What enabling this option really does is that |InnoDB| reads the data needed by the ``DML`` queries but does not actually update the records, and hence no undo log record is generated, as nothing has changed, which means that rollback is instantaneous, and |InnoDB| doesn't have to do any additional work on rollback. However, the problem of locking contention is not completely solved, when the records are read, SHARED locks are taken on the records, so this can still cause contention with data changes that SQL thread needs to make. |Percona Server| does have a variable :variable:`innodb_locking_fake_changes` to make fake changes implementation completely lock-less. Because the fake changes implementation is not ready for lock-less operation for all workloads this variable is not safe to use and that is why it is disabled by default.
+
+The :variable:`innodb_fake_changes` option only enables the behavior of transactions being rolled back on ``COMMIT``. It's by no way a tool that does prefetching of data. It merely provides a feature that is needed by prefetching tools to work in a performant manner. There is no prefetching that is transparently done by the slave when :variable:`innodb_fake_changes` is enabled, i.e., there is no change in slave behavior, there is no separate thread that is started to prefetch events. A separate utility is needed that runs with the session :variable:`innodb_fake_changes` variable enabled and that reads events from the relay log.
 
 Caveats
 =======
 
+.. warning:: 
+
+  This feature is only safe to use with a InnoDB-only server, because it is implemented in |InnoDB| only. Using it with any other storage engine such as |MyISAM| will cause data inconsistencies.
```

Comment:
MyISAM or TokuDB

... because COMMITs will not be rolled back on those storage engines